kafka-connect-aws-s3/src/test/scala/io/lenses/streamreactor/connect/aws/s3/sink/config/DeleteModeSettingTest.scala

```diff
@@ -0,0 +1,27 @@
+package io.lenses.streamreactor.connect.aws.s3.sink.config
+
+import com.typesafe.scalalogging.LazyLogging
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks._
+
+import scala.jdk.CollectionConverters.MapHasAsJava
+
+class DeleteModeSettingTest extends AnyFlatSpec with Matchers with LazyLogging {
+  it should "respect the delete mode setting" in {
+    val deleteModeMap = Table[String, String, Boolean](
+      ("testName", "value", "expected"),
+      ("all-at-once", "AllAtOnce", true),
+      ("individual", "Individual", false),
+    )
+
+    forAll(deleteModeMap) {
+      (name: String, value: String, expected: Boolean) =>
```

Comment:
Move the Table outside of the loop and use it to parameterise the test/set the test name

```suggestion
    
  private val deleteModeMap = Table[String, String, Boolean](
    ("testName", "value", "expected"),
    ("all-at-once", "AllAtOnce", true),
    ("individual", "Individual", false),
  )

  forAll(deleteModeMap) {
    (name: String, value: String, expected: Boolean) =>
    it should s"respect the delete mode setting $name" in {
```